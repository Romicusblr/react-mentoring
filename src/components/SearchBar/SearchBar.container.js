import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { fetchMovies } from '../../actions';
import SearchBar from './SearchBar';

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchMovies,
}, dispatch);

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SearchBar));
