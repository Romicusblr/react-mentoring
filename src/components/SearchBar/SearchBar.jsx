import React from 'react';
import ButtonRaised from '../ButtonRaised';
import SearchInput from '../SearchInput';
import RadioSwitch from '../RadioSwitch';
import RadioButton from '../RadioButton';
import style from './SearchBar.module.css';

const SearchBar = () => (
  <form className={style.searchBar}>
    <p>
      <SearchInput />
      <ButtonRaised>search</ButtonRaised>
    </p>
    <RadioSwitch name="searchFilter" title="search by">
      <RadioButton value="title" defaultChecked />
      <RadioButton value="genre" />
    </RadioSwitch>
  </form>
);

export default SearchBar;
