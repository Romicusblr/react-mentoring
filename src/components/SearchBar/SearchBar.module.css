.searchBar {
  display: flex;
  flex-direction: column;
  &>p {
    display: flex;
    margin-top: 0;
    &>input[type=text] {
      flex-grow: 1;
    }
  }
}
