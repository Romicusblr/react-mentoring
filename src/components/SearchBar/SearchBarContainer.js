import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchMovies } from '../../actions';
import SearchBar from './SearchBar';


const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchMovies,
}, dispatch);

const SearchBarContainer = connect(mapStateToProps, mapDispatchToProps)(SearchBar);

export default SearchBarContainer;
