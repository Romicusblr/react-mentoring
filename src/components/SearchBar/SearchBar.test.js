import SearchBar from '.';

describe('SearchBar', () => {
  it('should match to snapshot', () => {
    const snap = shallow(<SearchBar />);
    expect(snap).toMatchSnapshot();
  });
});
