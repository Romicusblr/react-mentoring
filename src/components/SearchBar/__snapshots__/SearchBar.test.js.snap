// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match to snapshot 1`] = `
<form
  className="searchBar"
>
  <p>
    <SearchInput
      className=""
      placeholder="type here"
    />
    <Button
      className=""
      type="submit"
    >
      search
    </Button>
  </p>
  <RadioSwitch
    className=""
    name="searchFilter"
    title="search by"
  >
    <RadioButton
      className=""
      defaultChecked={true}
      name="radio"
      value="title"
    />
    <RadioButton
      className=""
      defaultChecked={false}
      name="radio"
      value="genre"
    />
  </RadioSwitch>
</form>
`;
